packages/fxa-auth-server/lib/routes/recovery-key.js

```diff
@@ -249,6 +261,79 @@ module.exports = (log, db, Password, verifierVersion, customs, mailer) => {
         return db.recoveryKeyExists(uid);
       },
     },
+    {
+      method: 'GET',
+      path: '/recoveryKey/hint',
+      options: {
+        ...RECOVERY_KEY_DOCS.RECOVERYKEY_HINT_GET,
+        // No auth needed to fetch the recovery key hint
+        auth: false,
+        validate: {
+          query: {
+            uid: isA.string().min(32).max(32).regex(HEX_STRING),
```

Comment:
Q: Is there a difference between `.min(32).max(32)` and [`.length(32)`](https://joi.dev/api/?v=17.9.1#stringlengthlimit-encoding)?

```js
const isA = require("joi");

const str = "abcde";

const schema1 = isA.string().min(4).max(4).regex(/[a-z]/);
console.log(`"${str}": ${schema1.validate(str).error?.details.at(0).message ?? "valid"}`);

const schema2 = isA.string().length(4).regex(/[a-z]/);
console.log(`"${str}": ${schema2.validate(str).error?.details.at(0).message ?? "valid"}`);
```

```sh
"abc": "value" length must be at least 4 characters long
"abc": "value" length must be 4 characters long

"abcd": valid
"abcd": valid

"abcde": "value" length must be less than or equal to 4 characters long
"abcde": "value" length must be 4 characters long
```

Although, possibly moot since it looks like we might be able to claim "prior art" on the `.min().max()` syntax:
```sh
git grep ".string()" packages | grep ".min(" | grep ".max("  | cat

packages/fxa-auth-server/lib/routes/account.ts:            uid: isA.string().min(32).max(32).regex(HEX_STRING),
packages/fxa-auth-server/lib/routes/emails.js:            code: isA.string().min(32).max(32).regex(HEX_STRING).required(),
packages/fxa-auth-server/lib/routes/recovery-key.js:            uid: isA.string().min(32).max(32).regex(HEX_STRING),
packages/fxa-auth-server/lib/routes/validators.js:  return isA.string().regex(regex).min(8).max(len);
packages/fxa-content-server/app/scripts/lib/vat.js:Vat.register('codeChallenge', Vat.string().min(43).max(128));
```

Reply:
The outcome might be the same, but `length` does seem better even though there are existing instances of min/max.